File: src/main/java/net/thedragonteam/armorplus/client/gui/APTab.java
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.client.gui;

import net.minecraft.block.Block;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import net.thedragonteam.armorplus.ArmorPlus;
import net.thedragonteam.armorplus.registry.APItems;
import net.thedragonteam.armorplus.registry.ModBlocks;
import net.thedragonteam.armorplus.registry.ModItems;

import static net.thedragonteam.thedragonlib.util.ItemStackUtils.getItemStack;

/**
 * @author Sokratis Fotkatzikis - TheDragonTeam
 */
public class APTab extends CreativeTabs {

    private int tab;
    private String label;

    public APTab(int id, String modid, String label, int tab) {
        super(id, modid);
        this.tab = tab;
        this.label = label;
        this.setBackgroundImageName(ArmorPlus.MODID + ".png");// Automatically has tab_ applied to it. Make sure you change the texture name.
    }

    @Override
    public boolean hasSearchBar() {
        return true;
    }

    @Override
    @SideOnly(Side.CLIENT)
    public ItemStack getIconItemStack() {
        if (tab == 0) {
            return addTabItemStack(APItems.lavaChestplate);
        } else if (tab == 1) {
            return APItems.infusedLavaCrystal;
        } else if (tab == 2) {
            return addTabItemStack(ModBlocks.blockInfusedLavaCrystal);
        } else if (tab == 3) {
            return addTabItemStack(ModItems.battleAxe[7]);
        }
        return ItemStack.EMPTY;
    }

    private ItemStack addTabItemStack(Object stack) {
        return stack instanceof Block || stack instanceof Item ? !getItemStack(stack).isEmpty() ? getItemStack(stack) : getItemStack(ModItems.lavaCrystal) : ItemStack.EMPTY;
    }

    @Override
    @SideOnly(Side.CLIENT)
    public ItemStack getTabIconItem() {
        return this.getIconItemStack();
    }

    @Override
    public String getTabLabel() {
        return this.label;
    }
}

File: src/main/java/net/thedragonteam/armorplus/client/gui/APTab.kt
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.client.gui

import net.minecraft.creativetab.CreativeTabs
import net.minecraft.item.ItemStack
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly
import net.thedragonteam.armorplus.ArmorPlus
import net.thedragonteam.armorplus.registry.APItems
import net.thedragonteam.armorplus.registry.ModBlocks
import net.thedragonteam.armorplus.registry.ModItems

import net.thedragonteam.armorplus.registry.ModItems.lava
import net.thedragonteam.thedragonlib.util.ItemStackUtils.getItemStack

/**
 * net.thedragonteam.armorplus.client.gui
 * ArmorPlus created by sokratis12GR on 7/26/2016 4:42 PM.
 * - TheDragonTeam
 */
class APTab(id: Int, modid: String, private val label: String, private val tab: Int) : CreativeTabs(id, modid) {

    init {
        this.backgroundImageName = ArmorPlus.MODID + ".png" // Automatically has tab_ applied to it. Make sure you change the texture name.
    }

    override fun hasSearchBar(): Boolean {
        return true
    }

    @SideOnly(Side.CLIENT)
    override fun getIconItemStack(): ItemStack {
        when (tab) {
            0 -> return getItemStack(APItems.lavaChestplate)
            1 -> return APItems.enderDragonScale
            2 -> return getItemStack(ModBlocks.blockLavaCrystal)
            3 -> return getItemStack(APItems.redstoneBattleAxe)
            4 -> return getItemStack(ModItems.itemTeslaRod)
        }
        return ItemStack.EMPTY
    }

    @SideOnly(Side.CLIENT)
    override fun getTabIconItem(): ItemStack {
        return iconItemStack
    }

    override fun getTabLabel(): String {
        return this.label
    }

    companion object {

        lateinit internal var iconArmorPlus: ItemStack

        fun initialize() {
            iconArmorPlus = ItemStack(lava[1])
        }
    }
}